libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -209,6 +223,11 @@ - (SFSyncState*) syncDownWithTarget:(SFSyncDownTarget*)target options:(SFSyncOpt
 /** Resync
  */
 - (SFSyncState*) reSync:(NSNumber*)syncId updateBlock:(SFSyncSyncManagerUpdateBlock)updateBlock {
+    if (syncId.integerValue == self.syncIdRunning) {
```

Comment:
Are we sure about that?  I know that we use a serial queue to initiate the dispatch of a sync, but there are a number of asynchronous processes that happen after the initial dispatch—most notably an async network service that dispatches independently of the SmartSync dispatching.  The serial execution of a dispatch queue is only as serial as the synchronous code encapsulated within the dispatch.  I haven't had time to test it out, but my guess is that, once a network request has been queued for one sync, another sync will be freed to start.